src/classes/CRLP_RollupCMT.cls

```diff
@@ -408,6 +462,127 @@ public class CRLP_RollupCMT {
+            }
+            if (otherRollup.dateField != this.dateField) {
+                return true;
+            }
+            if (otherRollup.isActive != this.isActive) {
+                return true;
+            }
+            if (otherRollup.isDeleted != this.isDeleted) {
+                return true;
+            }
+            if (otherRollup.useFiscalYear != this.useFiscalYear) {
+                return true;
+            }
+            if (otherRollup.intValue != this.intValue) {
+                return true;
+            }
+
+            return false;
+        }
+
+        /**************************************************************************************************************
+        * @description Return the rollup type for a given rollup definition
+        * @param Rollup__mdt CMT Record
+        * @return CRLP_RollupProcessingOptions.RollupType
+        **************************************************************************************************************/
+        public CRLP_RollupProcessingOptions.RollupType getRollupType() {
+
+            String accountApiName = UTIL_Describe.getObjectDescribe('Account').getName();
+            String contactApiName = UTIL_Describe.getObjectDescribe('Contact').getName();
+            String opportunityApiName = UTIL_Describe.getObjectDescribe('Opportunity').getName();
+            String paymentApiName = UTIL_Describe.getObjectDescribe('npe01__OppPayment__c').getName();
+            String recurringDonationsApiName = UTIL_Describe.getObjectDescribe('npe03__Recurring_Donation__c').getName();
+            String allocationApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Allocation__c')).getName();
+            String partialSoftCreditApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Partial_Soft_Credit__c')).getName();
+            String accountSoftCreditApiName = UTIL_Describe.getObjectDescribe(UTIL_Namespace.StrAllNSPrefix('Account_Soft_Credit__c')).getName();
+
+            // work down from most specific
+            if (this.summaryObject == recurringDonationsApiName) {
+                return CRLP_RollupProcessingOptions.RollupType.RecurringDonations;
+            }
+
+            if (this.summaryObject == allocationApiName) {
+                return CRLP_RollupProcessingOptions.RollupType.GAU;
+            }
+
+            if (this.amountObject == partialSoftCreditApiName) {
+                if (this.summaryObject == accountApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit;
+                }
+                if (this.summaryObject == contactApiName) {
+                    return CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit;
+                }
+            }
+
+            if (this.amountObject == accountSoftCreditApiName) {
+                if (this.detailObject == opportunityApiName || this.detailObject == accountSoftCreditApiName) {
```

Comment:
Technically if the Amount Object is AccountSoftCredit then it's always an AccountSoftCredit rollup, no matter what the Detail object is. However, if there were a Count type rollup the Amount object might not be set (because it's ignored). I suggest using `detailObject == accountSoftCreditApiName || amountObject == accountSoftCreditApiName`.

Reply:
@patrick-yan-sf The second `if` is not needed here. If the amount or detail object is AccountSoftCredit, then it's always AccountSoftCredit so there's no need to check the detail object a second time.